### ISSUE TITLE
beat display to midi controller

### ISSUE THREAD

**Issue by georgkrause** [openAVproductions/openAV-Luppp#124]
What i want to do is this: i want that my midi controller (apc mini) displays the beat graph from the right bottom corner of luppp. i want to specify 4 buttons which shortly light up on the corresponding beat. So button 1 on beat 1, button 2 on beat to and so on. With #119 this would enable me to dont look at the screen anymore while making music, which would be great!

**Comment by georgkrause** [comment]
it would also be okay if just one button lights up on every beat...

**Comment by harryhaaren** [comment]
Mappa will handle user-defined feedback, allowing any "source" to be mapped to various lights. Aka, this should be possible then.

**Comment by wvengen** [comment]
I've implemented it in [this branch](https://github.com/wvengen/openAV-Luppp/tree/feature/midi-beat), which builds on PR #311. The controller JSON output binding would be like this (and one for each beat):
```json
{
  "action": "metronome:beat",
  "beat": 0,
  "dataList": [176, 99, 21]
}
```
Each beat (in the bar) would get its own controller output binding. Use of `dataList` would enable one to either use a single light with different colors (like above), or multiple lights (then the `dataList` would contain a message to clear all lights except the light of the active beat).

**Comment by harryhaaren** [comment]
Hi @wvengen : cool - nice one. I presume you want to turn off the LED too? A note-off will work for some devices, but not all... :/

**Comment by wvengen** [comment]
> I presume you want to turn off the LED too? A note-off will work for some devices, but not all... :/

Good point. The `dataList` can be of arbitrary size, so there can be as many MIDI events as you want, incl. note-off for all the beats. A later improvement could be to have on- and off-datalists, or beat-on and beat-off events, if this turns out to be a much-used feature.
